src/app/slices/settings.ts

```diff
@@ -0,0 +1,50 @@
+import type { PayloadAction } from "@reduxjs/toolkit"
+import { createSlice } from "codeforlife/slices"
+
+import { getSettingFromCookies } from "../util"
+
+export type TwoPanelLayout = "auto" | "vertical" | "horizontal"
+export type ThreePanelLayout =
+  | "auto"
+  | "verticalWithLeftHorizontal"
+  | "vertical"
+  | "horizontal"
+
+export interface SettingsState {
+  twoPanelLayout: TwoPanelLayout
+  threePanelLayout: ThreePanelLayout
+}
+
+const DEFAULT_SETTINGS: SettingsState = Object.freeze({
+  twoPanelLayout: "auto",
+  threePanelLayout: "auto",
+})
+
+const cookiesSettings = getSettingFromCookies()
+const initialState = cookiesSettings ? cookiesSettings : DEFAULT_SETTINGS
+
+// [!] This slice has additional side effects in
+// `settings-cookie-persistence` middleware
+const settingsSlice = createSlice({
+  name: "settings",
+  initialState,
```

Comment:
1. remove comments - middlewares are generally understood to have side effects
1. use nullish coalescing operator instead ternary operator
```suggestion
const settingsSlice = createSlice({
  name: "settings",
  initialState: getSettingsCookie() ?? DEFAULT_SETTINGS,
```